src/libasr/casting_utils.h

```diff
@@ -0,0 +1,21 @@
+#ifndef LFORTRAN_CASTING_UTILS_H
+#define LFORTRAN_CASTING_UTILS_H
+
+
+#include <libasr/asr.h>
+
+namespace LCompilers::CastingUtil {
+
+    int get_type_priority(ASR::ttypeType type);
+
+    int get_src_dest(ASR::expr_t* left_expr, ASR::expr_t* right_expr,
+                      ASR::expr_t*& src_expr, ASR::expr_t*& dest_expr,
+                      ASR::ttype_t*& src_type, ASR::ttype_t*& dest_type,
+                      bool is_assign);
+
+    ASR::expr_t* perform_casting(ASR::expr_t* expr, ASR::ttype_t* src,
+                                 ASR::ttype_t* dest, Allocator& al,
+                                 const Location& loc);
```

Comment:
What is the idea behind these? I think we do not want to do any kind of automatic casting in ASR, nor be responsible for determining what proper casting must be done.

Reply:
>I think we do not want to do any kind of automatic casting in ASR

It is actually not for automatic casting. The floordiv intrinsic we defined in this PR is generalized so that it can work on any type (from the supported types of `int, unsigned int, real, logical`). The casting function is used to cast the intermediate temporary variable to the appropriate return type.

Reply:
It is also used to cast the function arguments to the needed intermediate type.

Reply:
Is this a helper function to make `make_Cast_t_value` more convenient, but you still tell the initial and final type explicitly?

Reply:
>Is this a helper function to make make_Cast_t_value more convenient, but you still tell the initial and final type explicitly?

Yes. The way it helps is we need not worry about what type it currently is and just specify what type we want it to be.